mne/export/_edf.py

```diff
@@ -0,0 +1,184 @@
+        pmin, pmax = -3200, 3200
+
+        # check that physical min and max is not exceeded
+        if data.max() > pmax:
+            raise RuntimeError('The maximum uV of the data is more then pmax.')
+        if data.min() < pmin:
+            raise RuntimeError('The minimum uV of the data is less then pmin.')
+
+    # create instance of EDF Writer
+    hdl = EDFwriter(fname, EDFwriter.EDFLIB_FILETYPE_EDFPLUS, n_chs)
+
+    # set channel data
+    for ichan, ch in enumerate(ch_names):
+        if physical_range == 'auto':
+            # take the channel type minimum and maximum
+            ch_type = raw.get_channel_types(picks=ch)[0]
+            pmin, pmax = ch_types_phys_min[ch_type], ch_types_phys_max[ch_type]
+
+        if hdl.setPhysicalMaximum(ichan, pmax) != 0:  # noqa
+            raise RuntimeError("setPhysicalMaximum() returned an error")
+        if hdl.setPhysicalMinimum(ichan, pmin) != 0:  # noqa
+            raise RuntimeError("setPhysicalMinimum() returned an error")
+        if hdl.setDigitalMaximum(ichan, digital_max) != 0:  # noqa
+            raise RuntimeError("setDigitalMaximum() returned an error")
+        if hdl.setDigitalMinimum(ichan, digital_min) != 0:  # noqa
+            raise RuntimeError("setDigitalMinimum() returned an error")
+        if hdl.setPhysicalDimension(ichan, phys_dims) != 0:  # noqa
+            raise RuntimeError("setPhysicalDimension() returned an error")
+        if hdl.setSampleFrequency(ichan, sfreq) != 0:  # noqa
+            raise RuntimeError("setSampleFrequency() returned an error")
+        if hdl.setSignalLabel(ichan, ch) != 0:  # noqa
+            raise RuntimeError("setSignalLabel() returned an error")
+        # if hdl.setPreFilter(ichan, "HP:0.05Hz LP:40Hz N:60Hz") != 0:
+        #     raise RuntimeError("setPreFilter() returned an error")
+        #     sys.exit()
+        # if hdl.setTransducer(ichan, "AgAgCl cup electrode") != 0:
+        #     raise RuntimeError("setTransducer() returned an error")
+        #     sys.exit()
+
+    # set patient info
+    subj_info = raw.info.get('subject_info')
+    if subj_info is not None:
+        birthday = subj_info.get('birthday')
+        name = subj_info.get('first_name') + subj_info.get('last_name')
+        hand = subj_info.get('hand')
+        sex = subj_info.get('sex')
+
+        if birthday is not None:
+            if hdl.setPatientBirthDate(birthday[0], birthday[1], birthday[2]) != 0:  # noqa
+                raise RuntimeError("setPatientBirthDate() returned an error")
+        if hdl.setPatientName(name) != 0:  # noqa
+            raise RuntimeError("setPatientName() returned an error")
+        if hdl.setPatientGender(sex) != 0:  # noqa
+            raise RuntimeError("setPatientGender() returned an error")
+
+        if hdl.setAdditionalPatientInfo(f"hand={hand}") != 0:  # noqa
+            raise RuntimeError("setAdditionalPatientInfo() returned an error")
```

Comment:
this pattern repeats 11 times so far.  make a helper function that handles the error raising and replace these `if... raise` clauses with something like `try_to_set_value(hdl, 'AdditionalPatientInfo', f'hand={hand}')`

Reply:
Added a new private function